### ISSUE TITLE
Conflict with Airbnb guide in "arrow-parens" rule

### ISSUE THREAD

**Issue by apust** [iamturns/eslint-config-airbnb-typescript#19]
airbnb-typescript [uses](https://github.com/iamturns/eslint-config-airbnb-typescript/blob/master/lib/shared.js#L17) `as-needed` option for `arrow-parens` rule instead of `always` [suggested by Airbnb style guide](https://github.com/airbnb/javascript#arrows--one-arg-parens).

This is expected to be correct, but gets an error:
```JS
const myFunc = (singleArg) => singleArg * 2;
```

This is expected to be incorrect, but gets no error:
```JS
const myFunc = singleArg => singleArg * 2;
```

**Comment by jabacchetta** [comment]
Can confirm — this is a recent update to the style guide.

**Comment by JPeer264** [comment]
`arrow-parens` got changed because of https://github.com/iamturns/eslint-config-airbnb-typescript/issues/8

**Comment by deskoh** [comment]
@JPeer264 Verified that it's working with latest libs. See PR sergesemashko/airbnb-typescript-eslint-test#1. Rule should be enabled back.

**Comment by apust** [comment]
@deskoh Can't test your PR right now. Do we need to have only `requireForBlockBody`? Like that:
```JavaScript
"arrow-parens": [
  {
    requireForBlockBody: false,
  },
],
```

**Comment by deskoh** [comment]
> @deskoh Can't test your PR right now. Do we need to have only `requireForBlockBody`? Like that:
> 
> ```js
> "arrow-parens": [
>   {
>     requireForBlockBody: false,
>   },
> ],
> ```

The above rule is not valid. The rule can be removed as it is already specified by `airbnb`.

You can verify using `npx eslint --print-config test.js | grep -A 3 arrow-parens`.

**Comment by apust** [comment]
Can confirm: linting HOC as described in https://github.com/iamturns/eslint-config-airbnb-typescript/issues/8 no longer fails with latest style guide settings:
```JavaScript
"arrow-parens": [
  "error",
  "always",
]
```

I updated my PR https://github.com/iamturns/eslint-config-airbnb-typescript/pull/20

**Comment by deskoh** [comment]
Turns out this was no longer the issue because `airbnb` v18 and `airbnb-base` v14 `arrow-parens` rule no longer have `requireForBlockBody` option. See https://github.com/typescript-eslint/typescript-eslint/issues/14#issuecomment-500604282 for more info.

**Comment by iamturns** [comment]
Thank you @apust @jabacchetta @JPeer264 @deskoh - it's awesome to see this issue thoroughly researched, tested, and fixed entirely by the community! 👏 👏 👏
